### DISCUSSION TITLE
Handling empty  object in `useSearch` hook

### DISCUSSION THREAD

**Discussion by chiubaca** [TanStack/router#813] [Q&A]
I'm currently tinkering with a very simple mapping application and would like to sync the coordinates of the map with search params. Something slightly annoying which I'm not sure how handle is that the `useSearch` hook returns a union of an empty object and my expected type. e.g:

![image](https://github.com/TanStack/router/assets/18376481/55caaa96-c0a9-4509-be37-33924fa19264)

Here is how I have configured my route:

![image](https://github.com/TanStack/router/assets/18376481/4c43d9e9-60fd-4d02-a9a4-002da5901aed)

perhaps my mental model is incorrect. But I assumed that because I've validated the params in `validateSearch` we should be able assume  we _definitely_ have those search params available to use. I dont like the ergonomics of having to check the object is not empty every time.  Hope this all makes and someone can advise 🙏

**Comment by chiubaca** [comment]
Ah figure it out.

instead of:

```ts
const search = useSearch({from: '/ })
```

I had to do this instead:

```ts
import { indexRoute } from "../router";
/**
* export const indexRoute = new Route({
*  getParentRoute: () => rootRoute,
*  path: "/",
*  component: Map,
*  validateSearch: (search) => indexSearchSchema.parse(search),
});
**/


 const search = useSearch({ from: indexRoute.id });
```

and i could also refactor to this with TS moaning:

```ts
 const {lat, lng} = useSearch({ from: indexRoute.id });

```

**Reply by anjitpariyar** [reply]
isnt Importing indexRouter and writing manually '/' is same?
Im trying to create a custom hook , so the `from` value must be dynamic.

`
const useFilter = (from?: ValidRoutes) => {
 
  const search = useSearch({
    from: from,
    strict: false,
  })
`
like this but getting type error as , validRoutes is only '/dashboard' not '/_dasahboard/dashboard'

do you know any types that tanstackrouter provide that i can use?
